File: DIOT24/DIOT24h_moment_1/d2.1_to_5_prime/prime.c
#include <stdbool.h>

bool isPrime(long N) {
    /* Returns true if N is a prime number otherwise false */
    if(N < 2) return false;
    if(N == 2) return true;
    if(N % 2 == 0) return false;
    for(long i = 3; i < N; i++)
        if(N % i == 0) return false;
    return true;
}

File: MJU23-1/DSAlg/Course_DsAlg/D3/prime.cs
static bool IsPrime(int n)
{
    if (n < 0) return IsPrime(-n);
    if (n < 2) return false;
    for (int i = 2; i*i <= n; i++) { 
        if (n % i == 0) return false;
    }
    return true;
}